### PR TITLE
A few small fixes

### PR DESCRIPTION
**Pull request by abidlabs**
Fixes a few small things I noticed as I was building demos / writing the `Blocks` guide

* allow markdown to be set dynamically (i.e. used as an output component)
* provide aliases for components that have a 2-word name in case people use the wrong case (i.e. `gr.DataFrame` is now equivalent to `gr.Dataframe`) -- Closes: #1110
* allows you to set `inputs=None` or `outputs=None` to an event trigger
* removed a flaky test

### CODE REVIEW

File: gradio/__init__.py [hunk 1]
```diff
@@ -30,6 +34,7 @@
     Slider,
     StatusTracker,
     Textbox,
+    TimeSeries,
```
Comment (aliabid94) [inline gradio/__init__.py:37]: why twuce, typo?

File: gradio/interface.py [hunk 2]
```diff
@@ -166,12 +166,12 @@ def __init__(
         )
 
         self.interface_type = self.InterfaceTypes.STANDARD
-        if inputs is None and outputs is None:
+        if (inputs is None or inputs == []) and (outputs is None or outputs == []):
```
Comment (aliabid94) [inline gradio/interface.py:169]: since you call super.init() first, you could just check if self.inputs == [].

**Review by aliabid94 — APPROVED** [review]
LGTM, small comments

File: gradio/__init__.py (hunk 1 above)
Comment (abidlabs) [reply gradio/__init__.py:37]: Different spelling (or casing actually)

Comment (abidlabs) [reply gradio/__init__.py:37]: TimeSeries vs Timeseries

Comment (abidlabs) [reply gradio/__init__.py:37]: That is what you're referring to, right?

File: gradio/interface.py (hunk 2 above)
Comment (abidlabs) [reply gradio/interface.py:169]: Where in `Blocks()` do I assign `self.inputs`?

**Pull request merged**